packages/relay/src/lib/eth.ts

```diff
@@ -1281,14 +1281,6 @@ export class EthImpl implements Eth {
       throw predefined.INVALID_CONTRACT_ADDRESS(call.to);
     }
 
-    // If "From" is distinct from blank, we check is a valid account
```

Comment:
Q: Do you think we should totally remove it?
I think we should only remove it when the call has no value i.e. it's not for a transfer. In that case it doesn't matter. If the call has value then we should still check `from` fail fast and save us and mirror node the overhead of an unnecessary call.

Reply:
We could check the from address, if value is filled. But what do you think about removing all checks, `to` address field as well? Web3 module is currently doing all those check either way when we do the request and what we are doing it the relay is double work. So basically in all scenarios (we have not existing `from` or `to` or whatever) we are doing only one mirror-node request and that's it, compared to now -> in the best case 2 calls, in worst case 3 calls.
While keeping in mind, that we still have filed validations ( you can't pass incorrect address in `to`/`from` field, because it needs to be 20 bytes).

Reply:
Yeah i thin we should conditionally check the `from` field only if it's a `value` transfer case.

Reply:
Ah i think i'm seeing it now. I forgot the validations require a Mirror Node call.
okay, if we're calling the mirror node just to verify whether to call the mirror node when it's gonna make the same check then I agree let's not do double work in either `from` or `to` case.

Maybe move the check only to the fall back consensus node case

Reply:
I agree that we don't need to call 3 times the mirror node.

Reply:
I have moved the validation to `callConsensusNode`